packages/utils/src/ast.ts

```diff
@@ -389,7 +323,43 @@ export function postProcessAST(ast: Node): Node {
       path.replaceWith(replacement as Node);
       path.skip();
     },
-  });
 
-  return ast;
+    BinaryExpression(path) {
+      const source = path.getSource();
+
+      // Throw an error if HTML comments are used as a binary expression.
+      if (source.includes('<!--') || source.includes('-->')) {
+        throw new Error('HTML comments (`<!--` and `-->`) are not allowed.');
+      }
+    },
+  };
+
+  try {
+    const file = transformSync(code, {
```

Comment:
Is this preferred over using the parser and generator directly?

Reply:
Yes, it's explained in the PR description why.

Reply:
Yeah, I just meant in general, does Babel prefer that people do this over using the underlying APIs?